tests/lib/mem_alloc/testcase.yaml

```diff
@@ -10,7 +10,12 @@ tests:
     platform_exclude: twr_ke18f
     filter: TOOLCHAIN_HAS_NEWLIB == 1
     tags: clib newlib userspace
-  libraries.libc.newlibnano:
+  libraries.libc.newlibnano.gnuarmemb:
     extra_args: CONF_FILE=prj_newlibnano.conf
     toolchain_whitelist: gnuarmemb
     tags: clib newlib userspace
+  libraries.libc.newlibnano.zephyr:
+    arch_whitelist: arm
+    extra_args: CONF_FILE=prj_newlibnano.conf
+    toolchain_whitelist: zephyr
+    tags: clib newlib userspace
```

Comment:
`libc_nano` is not available in the Zephyr SDK at the moment because https://github.com/zephyrproject-rtos/sdk-ng/pull/102 has not been merged.

The tests "pass" now with Zephyr SDK pass because `NEWLIB_LIBC_NANO` config is not valid when `ZEPHYR_TOOLCHAIN_VARIANT` is `zephyr` and it effectively builds with just normal newlib (refer to the following code):

https://github.com/zephyrproject-rtos/zephyr/blob/3cc5bda2fa10b924c04dc58ff7055fa7bb683dcc/lib/libc/Kconfig#L43-L51

```
warning: NEWLIB_LIBC_NANO (defined at lib/libc/Kconfig:44) was assigned the value 'y' but got the
value 'n'. You can check symbol information (including dependencies) in the 'menuconfig' interface
(see the Application Development Primer section of the manual), or in the Kconfig reference at
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_NEWLIB_LIBC_NANO.html (which is
updated regularly from the master branch). See the 'Setting configuration values' section of the
Board Porting Guide as well.
```

IMHO, the following changes should be implemented, after https://github.com/zephyrproject-rtos/sdk-ng/pull/102 is merged and available in a Zephyr SDK release:

1. Remove `toolchain_whitelist` in testcase.yaml (or alternatively, update it to `toolchain_whitelist: zephyr gnuarmemb`)
2. Remove `if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb"` for `NEWLIB_LIBC_NANO` in lib/libc/Kconfig.

Reply:
Right, I missed checking the warning when building with zephyr toolchain :)
@stephanosio thanks for spotting this. So I'll just remove the last commit in this patch, and we should remember to apply the suggested changes once the sdk-ng patches are in.